### PR TITLE
ext_authz: make the ext_authz filter a dual filter

### PR DESCRIPTION
**Pull request by pianiststickman**
Used #23071 as a model.

Signed-off-by: Eugene Chan <eugenechan@google.com>

Risk Level: low
Testing: integration tested
Docs Changes: n/a
Release Notes: Makes the ext_authz filter a dual filter.
See also #10455

### CODE REVIEW

**Comment by pianiststickman** [comment]
cc @alyssawilk @paul-r-gall

File: test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc [hunk 1]
```diff
@@ -1006,6 +1006,86 @@ TEST_P(ExtAuthzLocalReplyIntegrationTest, DeniedHeaderTest) {
   cleanup();
 }
 
+TEST_P(ExtAuthzLocalReplyIntegrationTest, UpstreamTest) {
```
Comment (alyssawilk) [inline test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:1009]: would it be plausible to change the whole test to dual mode?

**Review by alyssawilk — COMMENTED** [review]
Very cool!

Can you do a scan of
source/docs/upstream_filters.md
And make sure you address any open issues?  For example a quick check of source/extensions/filters/http/ext_authz/ there's a call to downstreamCallbacks
For that we may need to do special handling to not crash, and also document in filter docs that the route will not be selected when changing headers in upstream filter mode.
/wait

File: test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc (hunk 1 above)
Comment (pianiststickman) [reply test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:1009]: This test is more or less the same as the existing downstream filter test. Are you suggesting to parameterize it?

Comment (alyssawilk) [reply test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc:1009]: right, I'm saying we could run all the tests in both modes and maybe flush out other bugs

**Pull request closed without merging**